sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/db/migration/V8520__Attachments.java

```diff
@@ -0,0 +1,173 @@
+      }
+    }
+
+    dropTempTable();
+  }
+
+  private void scanTypeDir(EntityType type, PreparedStatement insertStatement) throws SQLException {
+    File dir = new File(filesDir + type.getDirName());
+    if (dir.exists()) {
+      log.info(String.format("scanning directory %s", dir.getAbsolutePath()));
+      try (PreparedStatement getStatement = connection
+          .prepareStatement(String.format("SELECT * FROM %s WHERE %s = ?", type.getTable(), type.getPrimaryKey()));
+          PreparedStatement joinStatement = connection.prepareStatement(
+              String.format("INSERT INTO %s(%s, attachmentId) VALUES (?, ?);", type.getJoinTable(), type.getPrimaryKey()))) {
+        for (File typeFile : dir.listFiles()) {
+          processObjectDir(type, typeFile, getStatement, insertStatement, joinStatement);
+        }
+      }
+    }
+  }
+
+  private void processObjectDir(EntityType type, File dir, PreparedStatement getStatement, PreparedStatement insertStatement,
+      PreparedStatement joinStatement) throws SQLException {
+    log.info("scanning directory " + dir.getAbsolutePath());
+    if (!dir.isDirectory()) {
+      log.warning(String.format("Unexpected non-directory found: %s", dir.getAbsolutePath()));
+    } else {
+      long entityId = 0L;
+      try {
+        entityId = Long.parseLong(dir.getName());
+      } catch (NumberFormatException e) {
+        // ignore error - failure handled below
+      }
+      if (entityId == 0L) {
+        log.warning(String.format("Unexpected directory name: %s", dir.getAbsolutePath()));
+      } else {
+        getStatement.setLong(1, entityId);
+        if (!getStatement.executeQuery().next()) {
+          log.warning(String.format("Found files for non-existant %s %d", type.getTable(), entityId));
+        } else {
+          joinStatement.setLong(1, entityId);
+          for (File objectFile : dir.listFiles()) {
+            processFile(objectFile, insertStatement, joinStatement);
+          }
+        }
+      }
+    }
+  }
+
+  private void processFile(File objectFile, PreparedStatement insertStatement, PreparedStatement joinStatement) throws SQLException {
+    if (!objectFile.isFile()) {
+      log.warning(String.format("Unexpected non-file found: %s", objectFile.getAbsolutePath()));
+    } else {
+      log.info("processing file " + objectFile.getAbsolutePath());
+      Path filePath = Paths.get(objectFile.getAbsolutePath());
+      String relativePath = File.separator + basePath.relativize(filePath).toString();
+
+      insertStatement.setString(1, objectFile.getName());
+      insertStatement.setString(2, relativePath);
+      if (insertStatement.executeUpdate() == 0) {
+        throw new RuntimeException("Failed to insert attachment data");
+      }
+
+      joinStatement.setLong(2, getGeneratedId(insertStatement));
+      if (joinStatement.executeUpdate() == 0) {
+        throw new RuntimeException("Failed to insert attachment relationship data");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)